packages/calypso-e2e/src/lib/pages/marketing-page.ts

```diff
@@ -66,4 +83,69 @@ export class MarketingPage {
 		await this.page.click( selectors.seoPreviewPaneCloseButton );
 		await this.page.waitForSelector( selectors.seoPreviewButton );
 	}
+
+	/* Social Connectisons */
+
+	/**
+	 * Clicks on the Connection button for specified Social service.
+	 *
+	 * @param {SocialConnection} target Social service.
+	 */
+	async clickSocialConnectButton( target: SocialConnection ): Promise< Page > {
+		// Set up a handler for the popup promise.
+		const popupPromise = this.page.waitForEvent( 'popup' );
+
+		await this.page
+			.getByRole( 'main' )
+			.getByRole( 'listitem' )
+			.filter( { hasText: target } )
+			.getByRole( 'button', { name: 'Connect' } )
+			.click();
+
+		return await popupPromise;
+	}
+
+	/**
+	 * Validates the specified Social service button now is connected.
+	 *
+	 * @param {SocialConnection} target Social service.
+	 */
+	async validateSocialConnected( target: SocialConnection ): Promise< boolean > {
+		try {
+			await this.page
+				.getByRole( 'main' )
+				.getByRole( 'listitem' )
+				.filter( { hasText: target } )
+				.getByRole( 'button', { name: 'Disconnect' } )
+				.waitFor();
+			return true;
+		} catch {
+			return false;
+		}
+	}
```

Comment:
**Nit (non-blocking):** In our previous test design patterns, [and even in our docs](https://wpcalypso.wordpress.com/devdocs/test/e2e/docs/patterns_tricks_gotchas.md), `validate` functions throw instead of returning booleans, so I think that might be more consistent to do here if you're okay with it!

Reply:
Good reminder, thank you!